### PR TITLE
Fix leading trivia is not applied if raw syntax is empty

### PR DESCRIPTION
**Pull request by kimdv**
Not sure if we should create a test class for `RawSyntax` where we can add the test case, or it is enough with the current ones

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci Please test

File: Sources/SwiftSyntax/SyntaxData.swift [hunk 1]
```diff
@@ -356,10 +356,18 @@ struct SyntaxData {
   }
 
   func withLeadingTrivia(_ leadingTrivia: Trivia) -> SyntaxData {
-    return replacingSelf(raw.withLeadingTrivia(leadingTrivia))
+    if let raw = raw.withLeadingTrivia(leadingTrivia) {
+      return replacingSelf(raw)
+    } else {
+      return replacingSelf(raw)
```
Comment (ahoppen) [inline Sources/SwiftSyntax/SyntaxData.swift:362]: I think in this case calling `replacingSelf(raw)` is a no-op so we can just `return self` here. Same below.

File: Sources/SwiftSyntax/RawSyntax.swift [hunk 2]
```diff
@@ -1113,7 +1113,7 @@ final class RawSyntax: ManagedBuffer<RawSyntaxBase, RawSyntaxDataElement> {
     }
   }
 
-  func withLeadingTrivia(_ leadingTrivia: Trivia) -> RawSyntax {
+  func withLeadingTrivia(_ leadingTrivia: Trivia) -> RawSyntax? {
```
Comment (ahoppen) [inline Sources/SwiftSyntax/RawSyntax.swift:1116]: Could you add documentation here what returning `nil` means? I’d suggest
```swift
/// Replaces the leading trivia of the first token in this syntax tree by `leadingTrivia`.
/// If the syntax tree did not contain a token and thus no trivia could be attached to it, `nil` is returned.
```

Similar below.

File: Tests/SwiftSyntaxBuilderTest/FunctionTests.swift [hunk 3]
```diff
@@ -52,7 +52,7 @@ final class FunctionTests: XCTestCase {
     let syntax = buildable.buildSyntax(format: Format(), leadingTrivia: leadingTrivia)
 
     XCTAssertEqual(syntax.description, """
-      func fibonacci(_ n: Int)-> Int{
+      ␣func fibonacci(_ n: Int)-> Int{
```
Comment (ahoppen) [inline Tests/SwiftSyntaxBuilderTest/FunctionTests.swift:55]: Haha, good catch 👍

**Review by ahoppen — COMMENTED** [review]
Thanks. Looks good to me. I think the current test coverage is fine. AFAICT we don’t have any tests at the `RawSyntax` level at the moment.

File: Sources/SwiftSyntax/RawSyntax.swift (hunk 2 above)
Comment (kimdv) [reply Sources/SwiftSyntax/RawSyntax.swift:1120]: Added! 
Also added parameter documentation

**Comment by kimdv** [comment]
@swift-ci Please test

**Review by ahoppen — APPROVED** [review]

**Pull request merged**